source/taskmanager/src/main/java/nl/aerius/taskmanager/TaskConsumer.java

```diff
@@ -110,12 +114,21 @@ public void messageDeliveryAborted(final Message<MessageMetaData> message, final
     taskMessageHandler.messageDeliveryAborted(message, exception);
   }
 
-  public void start() {
-    executorService.submit(() -> {
+  public synchronized void start() {
+    if (messageHandlerFuture != null && !messageHandlerFuture.isDone()) {
+      try {
+        messageHandlerFuture.get();
+      } catch (final InterruptedException e) {
+        Thread.currentThread().interrupt();
+      } catch (final ExecutionException e) {
+        LOG.trace("TaskConsumer shutdown {} got exception.", taskQueueName, e);
+      }
+    }
+    messageHandlerFuture = executorService.submit(() -> {
       try {
         taskMessageHandler.start();
       } catch (final IOException e) {
-        LOG.error("TaskConsumer for {} got IO problems.", taskQueueName, e);
+        LOG.trace("TaskConsumer for {} got IO problems.", taskQueueName, e);
```

Comment:
Any particular reason to change this to a trace? Wouldn't you want to know about this when it happens (though looking at `RabbitMQMessageHandler` it won't be thrown anyway).

Reply:
Yes this should be back to error. With work in process I changed it as it was covered elsewhere, but that code didn't make it. Therefor it should be as before.